### PR TITLE
Add `ComponentProps` util

### PR DESCRIPTION
**Pull request by Princesseuh**
## Changes

Add a little shortcut to `Parameters<typeof Component>[0]` for people who want to get the props of a component. Other framework have similar utilities (React and Svelte at least), so figured we can as well.

## Testing

N/A, we don't really have tests for type utils

## Docs

[Added to a PR that updates the TypeScript page](https://github.com/withastro/docs/pull/6702)

### CODE REVIEW

File: .changeset/cuddly-moons-hang.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
+"astro": minor
+---
+
+Adds a new `ComponentProps` type export from `astro/types` to get the props type of an Astro component.
+
+```astro
+---
+import { ComponentProps } from 'astro/types';
```
Comment (natemoo-re) [inline .changeset/cuddly-moons-hang.md:9]: Maybe just for clarity?

```suggestion
import type { ComponentProps } from 'astro/types';
```

**Review by natemoo-re — APPROVED** [review]
LGTM!

File: .changeset/cuddly-moons-hang.md (hunk 1 above)
Comment (Princesseuh) [reply .changeset/cuddly-moons-hang.md:9]: Yep, definitely!

**Review by bluwy — APPROVED** [review]

**Review by sarah11918 — APPROVED** [review]
Changeset LGTM! :partying_face:

**Pull request merged**